### PR TITLE
Correct AABB 'grow_by'

### PR DESCRIPTION
**Pull request by malbach**
Actual implementation makes AABB grow by twice the requested amount.

### CODE REVIEW

File: core/math/aabb.cpp [hunk 1]
```diff
@@ -54,8 +54,8 @@ void AABB::merge_with(const AABB &p_aabb) {
 
 	beg_1 = position;
 	beg_2 = p_aabb.position;
-	end_1 = Vector3(size.x, size.y, size.z) + beg_1;
-	end_2 = Vector3(p_aabb.size.x, p_aabb.size.y, p_aabb.size.z) + beg_2;
+	end_1 = beg_1 + size;	//	Vector3(size.x, size.y, size.z) + beg_1;
```
Comment (akien-mga) [inline core/math/aabb.cpp]: Please remove the comments, it's dead code.

File: core/math/aabb.h [hunk 2]
```diff
@@ -355,12 +355,13 @@ bool AABB::smits_intersect_ray(const Vector3 &p_from, const Vector3 &p_dir, real
...
-	size.x += 2.0 * p_amount;
-	size.y += 2.0 * p_amount;
-	size.z += 2.0 * p_amount;
+	position.x -= 0.5 * p_amount;
+	position.y -= 0.5 * p_amount;
+	position.z -= 0.5 * p_amount;
+	size.x += p_amount;
+	size.y += p_amount;
+	size.z += p_amount;
+
```
Comment (akien-mga) [inline core/math/aabb.h]: `clang-format` doesn't like this empty line before the closing bracket :)
https://travis-ci.org/godotengine/godot/jobs/417196721

Comment (malbach) [reply core/math/aabb.h]: Yep, I knew it and I'm using Godot's clang rules in VS2017, but looks like the latest update (VS2017 v15.8) broke something as I'm having problems with others projects...
As usual, I always forget to wait before updating any Micro$oft product...

**Comment by Zylann** [comment]
Currently the amount is how much on each side you want it to grow, I wonder if such a breaking change is really necessary.

Also it will potentially break all places where it is used, such as:

- aabb.cpp
- geometry.cpp
- rasterizer_storage_gles3.cpp
- spatial_editor_plugin.cpp
- csg.cpp
- voxel_light_baker.cpp
- visual_server_scene.cpp

**Comment by groud** [comment]
I think the API is better as is. For consistency with the 2D part.

**Comment by malbach** [comment]
@Zylann OK I see, thanks for the explanation, I think I misread the doc 😖 
and after seeing all that, I agree with you and @groud, this PR brings too much changes for nearly no gain...
I close the request

**Comment by groud** [comment]
No problem :) thanks for contributing.

**Pull request closed without merging**